api/v1alpha1/hostedcluster_types.go

```diff
@@ -146,6 +146,7 @@ type HostedClusterSpec struct {
 	// specified, an ID is generated. After initial creation, the value is
 	// immutable.
 	// +kubebuilder:validation:Pattern:="[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="clusterID is immutable"
```

Comment:
I think we also need a type scoped rule here to validate once set it's not removed, e.g. https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutablility-after-first-modification

Reply:
Ah yes, actually I ran into that with the webhooks too.  Good catch.